src/Features/Blockcore.Features.Wallet/Database/WalletStore.cs

```diff
@@ -224,58 +220,46 @@ public IEnumerable<TransactionOutputData> GetUnspentForAddress(string address)
 
         public WalletBalanceResult GetBalanceForAddress(string address, bool excludeColdStake)
         {
-            string excludeColdStakeSql = excludeColdStake && this.network.Consensus.IsProofOfStake ? "AND IsColdCoinStake != true " : string.Empty;
+            var transactionGroups = this.trxCol.Find(Query.All())
+              .Where(x => x.Address == address)
+              .Where(x => x.SpendingDetails == null)
+              .Where(x => excludeColdStake && this.network.Consensus.IsProofOfStake ? (x.IsColdCoinStake != true) : true)
+              .GroupBy(x => x.BlockHeight != null)
+              .ToList();
 
-            var sql = "SELECT " +
```

Comment:
How confident are you about this?
I am not sure we have tests here.

Reply:
Give me a day to run through some manual testing, I will let you know when it is ready.

Reply:
Hi @dangershony, this is ready as I've made the final adjustments. I've done all of my manual tests, but I cannot run the automated unit tests because the ReadyData needs to be regenerated.

> How confident are you about this?
> I am not sure we have tests here.
There are a couple tests for this method, but we need the readydata to be regenerated for it.